blocklylib-core/src/main/java/com/google/blockly/android/clipboard/SingleMimeTypeClipDataHelper.java

```diff
@@ -0,0 +1,116 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.blockly.android.clipboard;
+
+import android.content.ClipData;
+import android.content.ClipDescription;
+import android.content.Context;
+import android.content.Intent;
+import android.view.DragEvent;
+
+import com.google.blockly.android.R;
+import com.google.blockly.android.control.BlocklyController;
+import com.google.blockly.android.ui.PendingDrag;
+import com.google.blockly.android.ui.WorkspaceHelper;
+import com.google.blockly.model.Block;
+import com.google.blockly.utils.BlocklyXmlHelper;
+
+import java.io.IOException;
+
+/**
+ * Implements ClipDataTransformer with a single support MIME type.  Uses intent extras as the
+ * in-transit storage format.
+ */
+public class SingleMimeTypeClipDataHelper implements BlockClipDataHelper {
+    public static final String EXTRA_BLOCKLY_XML = "BLOCKLY_XML";
+
+    /**
+     * This constructs a {@link SingleMimeTypeClipDataHelper} with a MIME type derived
+     * from the application's package name. This assumes all Blockly workspaces in an app use with
+     * a shared set of blocks, and blocks can be dragged/copied/pasted them, even if they are
+     * in different Activities. It also ensures blocks from other applications will be rejected.
+     *
+     * @param context
+     * @return
+     */
+    public static BlockClipDataHelper getDefault(Context context) {
+        String mimeType = "application/x-blockly-" + context.getPackageName() + "+xml";
+        return new SingleMimeTypeClipDataHelper(mimeType, R.string.blockly_clipdata_label_default);
+
+    }
+
+    protected final String mMimeType;
+    protected final int mClipLabelRes;  // TODO(#): Singular vs plural ("block" vs "blocks")
+
+    protected BlocklyController mController;
+    protected WorkspaceHelper mViewHelper;
+    protected Context mContext;
+    protected String mClipLabel;
+
+    /**
+     * Constructs a new {@link SingleMimeTypeClipDataHelper} with the provided MIME string and
+     * user visible (accessibility, etc.) clip label string.
+     *
+     * @param mimeType The MIME type the new instance use for encoding and decoding.
+     * @param clipLabelRes The resource id of the label to apply to {@link ClipData}s.
+     */
+    public SingleMimeTypeClipDataHelper(String mimeType, int clipLabelRes) {
+        mMimeType = mimeType;
+        mClipLabelRes = clipLabelRes;
+    }
+
+    @Override
+    public void setController(BlocklyController controller) {
```

Comment:
Only the Context is used. Why not just take the Context in the constructor, or take the label instead of the resId and then not even the Context is needed outside the static constructor?